### ISSUE TITLE
Bug: Random preloads added for images

### ISSUE THREAD

**Issue by muteor** [facebook/react#27910]
React version: 18.3.0-canary-c29ca23af-20231205

## Steps To Reproduce

1. Render SSR app that has images in its tree using renderToString or renderToPipeableStream

Link to code example:

## The current behavior

![image](https://github.com/facebook/react/assets/305164/2cb278b5-bdd0-43e4-9252-6e054a9481c6)

## The expected behavior

To not insert preloads randomly.

Note that there is difference between renderToString and renderToPipeableStream, renderToString the preloads appear inside root and renderToPipeableStream they appear in head.

Sorry this is a bit of a question rather than bug report maybe, I have searched around and cannot see any mention of auto-preloading images but I assume there must be something going on that I have missed. 

I was able to fix by using `18.3.0-canary-493f72b0a-20230727`.

**Comment by nickluger** [comment]
This. Tested with `19.0.0-canary-a73c3450e-20240329`.

It would be "kinda" tolerable for the LCP image above fold (featured image or similar), but we have lots of lazy loaded images below fold which contain <noscript> alternatives like:
```tsx
<img
  {...imgProps}
/>
<noscript>
  <img
    alt={alt}
    src={noscriptSrc}
  />
</noscript>
```

For every `<img>` tag used in noscript, the image is preloaded in `<head>` which completely defies the purpose of lazy loading.

Beyond that, React shouldn't inject anything into my head arbitrarily.

**Comment by muteor** [comment]
I think this is actually now expected behaviour, after digging the source a bit I found https://github.com/facebook/react/commit/f359f9b41ac1a8127f5ba505e0c04675eee0d310

Though I could not find any docs/release notes that mention this which is a little odd as now you need to explicitly think about the loading strategy of every img tag you use. For me I was able to "fix" this by making sure all our img tags have either loading=lazy or fetchPriority=low when they are not to be preloaded, this gave me more or less only the LCP images preloaded.

@nickluger For your case you could either add loading=lazy or fetchPriority=low to the fallback images, though depending on your target browsers you could get rid of the noscript parts these days as loading=lazy has pretty wide browser support now.

Would be good to have this documented somewhere and also maybe add this to the react linter.

**Comment by nickluger** [comment]
Thanks for the info regarding loading=lazy, that could solve it. 🙏

If this is really a "feature", I have to say, I don't think it should be React's business to put stuff into my `<head>` automagically…

**Comment by eps1lon** [comment]
@muteor Which framework are you using?

**Comment by nickluger** [comment]
For me, its Remix.

**Comment by eps1lon** [comment]
You're probably using their image component that uses `ReactDOM.preload`. Or they're doing `ReactDOM.preload` calls some other way. Can you open an issue on their issue tracker first to clarify?

**Comment by gnoff** [comment]
>For every <img> tag used in noscript, the image is preloaded in <head> which completely defies the purpose of lazy loading.

This is a genuine bug

**Comment by nickluger** [comment]
Good to know, thanks for the explanation. 👍

**Comment by muteor** [comment]
fwiw no framework used just renderToPipeableStream, in my case I was able to add low priority to a few icon images to get the preload to load reasonable candidates.

Would be good to be able to opt-out of this somehow for a boundary, thinking of the case where you are waiting on component/data to load and don't really care about the image but lazy loading would be inappropriate for an above the fold larger image. Maybe even a way to modify the heuristic via some sort of callback would be useful.

**Comment by imjordanxd** [comment]
I recently ran into this issue as well for snapshot tests. It'd be good if this change was documented somewhere 👍

**Comment by jamesgpearce** [comment]
This is _very_ surprising behavior. Worse that it's not documented nor optional.

Especially since if you have already explicitly added the preload link tags to the head yourself and then boom... now you have two of them in the DOM.

No framework; just a raw `renderToStaticMarkup` will do this.

**Comment by talatkuyuk** [comment]
I agree with @jamesgpearce, 

I also encountered this issue while testing the parsing of a simple MDX content that only contains an image syntax.

I saw `<link rel="preload" as="image" href="image.png"/>` in the result . Then, while digging where does it come from, then I dropped here. 

I just used a raw `renderToStaticMarkup` in `react-dom@19`, `react-dom/server` in my test. I confirm `react-dom@18` doesn't have this behavior.

**Now, I want to know how to disable it using an option.**

**Comment by errorx666** [comment]
I'm using `renderToStaticMarkup` to render HTML to a forum.  It supports limited HTML, which does not include `link` tags.  Any time I include an `img` tag, I get an unwanted `link` tag to go with it.

It is surprising and problematic for me that it renders any markup I didn't explicitly specify, and worse, there's no obvious way to opt-out.

**Comment by brunolemos** [comment]
If you are calling `renderToStaticMarkup` outside a lifecycle method, you can use this instead:

```js
import { createRoot } from 'react-dom/client';
import { flushSync } from 'react-dom';

export function reactToString(element: React.ReactNode): string {
  const div = document.createElement('div');
  const root = createRoot(div);

  flushSync(() => {
    root.render(element);
  });

  return div.innerHTML;
}
```

If not, you can remove the link tag from the generated html, as a workaround:

```js
import { renderToStaticMarkup } from 'react-dom/server';

export function reactToString(element: React.ReactNode): string {
  let html = renderToStaticMarkup(element);

  html = html.replace(
    /<link\s+rel="preload"\s*as="image"\s*href="[^"]+"\s*\/>/g,
    '',
  );

  return html;
}
```
